src/interfaces/parser.ts

```diff
@@ -112,6 +112,16 @@ export type FlagProps = {
   hidden?: boolean;
   required?: boolean;
   dependsOn?: string[];
+  relationships?: {
+    dependsOn?: {
+      type: 'all' | 'atLeastOne';
+      flags: string[];
+    };
+    exclusive?: {
+      type: 'all' | 'atLeastOne';
```

Comment:
checking my understanding:

'all' => it's an error if you use all the flags listed?
'atLeastOne' => it's an error if you use any of the flags listed?